src/main/java/org/jboss/forge/addon/swarm/facet/impl/WildflySwarmFacetImpl.java

```diff
@@ -126,4 +139,57 @@ public void setConfiguration(WildflySwarmConfiguration configuration)
    {
       this.configuration = configuration;
    }
+   
+   @Override
+   public void installFractions(Iterable<String> selectedFractions)
+   {
+      Iterator<String> iterator = selectedFractions.iterator();
+      while (iterator.hasNext()) {
+         String[] parts = iterator.next().split(":");
+         Dependency dependency = DependencyBuilder.create(parts[0]).setArtifactId(parts[1])
+               .setVersion("${version.wildfly-swarm}");
+         DependencyFacet facet = getFaceted().getFacet(DependencyFacet.class);
+         facet.addDirectDependency(dependency);
+      }
+   }
+   
+   @Override
+   public List<String> getFractionList()
+   {
+      Dependency dependency = dependencyResolver.resolveArtifact(DependencyQueryBuilder.create(
+            CoordinateBuilder.create().setGroupId("org.wildfly.swarm").setArtifactId("wildfly-swarm-fraction-list")
+                  .setVersion("1.0.0.Alpha6-SNAPSHOT").setPackaging("txt")));
+
+      MavenFacet maven = getFaceted().getFacet(MavenFacet.class);
+      Model pom = maven.getModel();
+      Scanner s;
+      List<String> list = new ArrayList<String>();
+      try {
```

Comment:
Put the `Scanner` in this try-with-resources block: 

`try (Scanner sc = ...) {...}`